### ISSUE TITLE
Helm : no template "minio.serviceAccountName" associated with template "gotpl"

### ISSUE THREAD

**Issue by olevitt** [minio/minio#13199]
Hi,

Since helm chart `3.1.0`, helm installation fails with  
```
Error: template: minio/templates/serviceaccount.yaml:5:11: executing "minio/templates/serviceaccount.yaml" at <include "minio.serviceAccountName" .>: error calling include: template: no template "minio.serviceAccountName" associated with template "gotpl"
```  

Steps to reproduce :  
```
helm install --namespace minio --set rootUser=rootuser,rootPassword=rootpass123 --generate-name minio/minio
```  
(I do believe every install on `3.1.0` will fail, as long as ` .Values.serviceAccount.create` is not explicitly set to `false`)

Previous version `3.0.2` worked fine.  
It seems to be related to the new `helm/minio/templates/serviceaccount.yaml` file which refers to a non-existing `{{ include "minio.serviceAccountName" . | quote }}` (see commit https://github.com/minio/minio/commit/410b8dd0fd0e8a7968e787fb245848cdb9ceab43).  

I think we should add the corresponding tpl function to the `_helpers.tpl` file.   
For example, the one created by `helm create` :  
```
{{/*
Create the name of the service account to use
*/}}
{{- define "minio.serviceAccountName" -}}
{{- if .Values.serviceAccount.create }}
{{- default (include "minio.fullname" .) .Values.serviceAccount.name }}
{{- else }}
{{- default "default" .Values.serviceAccount.name }}
{{- end }}
{{- end }}
```

**Comment by klauspost** [comment]
See https://github.com/minio/minio/pull/13197

**Comment by harshavardhana** [comment]
fixed by #13197
